src/aznfswatchdog

```diff
@@ -122,20 +220,80 @@ reconcile_conntrack()
             if [[ "$line" =~ $matchstr ]]; then
                 l_sport=${BASH_REMATCH[1]}
                 l_dport=${BASH_REMATCH[2]}
-                reconcile_conntrack_one $l_ip $l_sport $l_dport $l_nfsip
+                reconcile_conntrack_synsent $l_ip $l_sport $l_dport $l_nfsip
+            fi
+        done
+    fi
+
+    IFS=$'\n' output111=$(conntrack -L -p tcp -d $l_ip --dport 111 | grep -v "SYN_SENT" | grep "\[UNREPLIED\]" 2>/dev/null)
+    IFS=$'\n' output2048=$(conntrack -L -p tcp -d $l_ip --dport 2048 | grep -v "SYN_SENT" | grep "\[UNREPLIED\]" 2>/dev/null)
+    output="$output111"$'\n'"$output2048"
+    if [ -n "$output" ]; then
+        for line in $output; do
+            # XXX Remove this log after running for few days.
+            vecho "$line"
+
+            matchstr=".* dst=$l_ip sport=([0-9]+) dport=([0-9]+) \[UNREPLIED\] src=([0-9]+\.[0-9]+\.[0-9]+\.[0-9]+).*"
+            if [[ "$line" =~ $matchstr ]]; then
+                l_sport=${BASH_REMATCH[1]}
+                l_dport=${BASH_REMATCH[2]}
+                l_reply_srcip=${BASH_REMATCH[3]}
+                reconcile_conntrack_unreplied $l_ip $l_sport $l_dport $l_reply_srcip
             fi
         done
     fi
 
     #
     # Any port that is not stuck now, means it's recovered since the last time and can be removed
-    # from the global cthash_synsent map.
+    # from the global cthash_synsent and cthash_unreplied map.
     #
     for key in "${!cthash_synsent[@]}"; do
         if [ ! -v cthash_synsent_now[$key] ]; then
             unset cthash_synsent[$key]
         fi
     done
+
+    for key in "${!cthash_unreplied[@]}"; do
+        if [ ! -v cthash_unreplied_now[$key] ]; then
+            unset cthash_unreplied[$key]
+        fi
+    done
+}
+
+#
+# Function for running stat for mountpoint so that everytime DNAT rule is updated, it is
+# used to make sure to send atleast one packet that matches the DNAT rule.
+# This will make sure that the connection gets a TCP reset and outstanding NFS RPC requests
+# are retransmitted right away w/o waiting for the 1 min timeout.
+#
+ping_new_endpoint()
+{
+    local target="$1"
+
+    stat "$target"
+    sleep 35
+    # One more stat after 30 sec sleep to let dir attributes timeout.
+    stat "$target"
+}
+
+run_daily_once()
+{
+    next_daily_run_epoch=$(expr $(date +%s) + $UPDATE_DETECTION_FREQUENCY)
+    
+    #
+    # To increase resilience in case the account is marked as regional due to a bad state,
+    # the arrays have been cleared. This ensures that we can check and mark the regional account on a daily basis.
+    # This helps prevent falsely marking an account as a regional account.
+    #
+    ip_change_count=()
+    last_ip_change_time=()
+    regional_accounts=()
+
+    # Since AKS users don't require auto-update, there's no need to check for updates.
+    if [ "$AKS_USER" == "true" ]; then
+        return 1
+    fi
+    check_for_latest_update=true
```

Comment:
since this AKS check is specifically for update, better to do

if [ "$AKS_USER" != "true" ]; then
    check_for_latest_update=true
fi